src/jit/lower.cpp

```diff
@@ -628,20 +620,14 @@ GenTree* Lowering::LowerSwitch(GenTree* node)
 
                 // Now, build the conditional statement for the current case that is
                 // being evaluated:
-                // GT_JTRUE
-                //   |__ GT_COND
-                //          |____GT_EQ
-                //                 |____ (switchIndex) (The temp variable)
-                //                 |____ (ICon)        (The actual case constant)
-                GenTreePtr gtCaseCond =
-                    comp->gtNewOperNode(GT_EQ, TYP_INT, comp->gtNewLclvNode(tempLclNum, tempLclType),
-                                        comp->gtNewIconNode(i, TYP_INT));
-                /* Increment the lvRefCnt and lvRefCntWtd for temp */
+                // GT_CMP switchValue, caseValue
+                // GT_JCC(EQ)
```

Comment:
> Question: when you say you have the bit test of 64 bit "flags" working, does that mean you're recognizing trees that collapse into a test of a single bit and turning them into bt?

The JIT already does an AND-CMP-to-TEST transform like the one you show in your example in `LowerCompare`. Where `BT` comes in handy is when you have an uncontainable immediate:
```asm
; if ((x & (1L << 42)) != 0) currently generates
       48B80000000000040000 mov      rax, 0x40000000000
       4885D0               test     rdx, rax
       740D                 je       SHORT G_M55888_IG03
; with BT this becomes
       0FBAE22A             bt       edx, 42
       770D                 ja       SHORT G_M55888_IG03
; and save 8-9 bytes of code
```
This is a kind of transform that cannot be done properly during codegen. Since the immediate value is not contained a register is allocated to it and the `mov` is generated before we get to see the `test` node and do something about it.